msteams-platform/sbs-botbuilder-conversation-AI.yml

```diff
@@ -0,0 +1,709 @@
+        ```
+
+        Your default web browser opens to let you sign in to the account. Sign in to your Azure account using your credentials. Close the browser when you're prompted.
+
+    2. Sign in to Azure with the TeamsFx CLI:
+
+        ``` bash
+        teamsfx account login azure
+        ```
+
+        Your default web browser opens to let you sign in to the account. Sign in to your Azure account using your credentials. Close the browser when you're prompted.
+
+        The account logins are shared between Visual Studio Code and the TeamsFx CLI.
+
+        Now that the development environment is configured, you can create, build, and deploy your first Teams app.
+
+    ---
+
+    ## Update your bot app with Conversational AI capabilities
+
+    Teams Toolkit is the easiest way to scaffold and get a bot running, Teams Toolkit creates the basic bot structure for you so that you can  get started with configuring your bot. 
+    
+    Let's get started to configure Teams AI library capabilities.
+
+    1. Go to Visual Studio Code and open a terminal window.
+    
+    1. Run the following command in the terminal to add the latest version of the Teams AI library library to your application using npm:
+
+       ```bash
+       npm install @microsoft/botbuilder-m365
+
+       ```
+
+    1. Go to the **env** folder and update the Open AI key in the `.env.local.user`file.
+
+       ```
+        SECRET_BOT_PASSWORD=
+        SECRET_OPENAI_API_KEY=sk-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+       ```
+
+    1. Go to the `config.ts` file and add the following code:
+    
+       ```typescript
+       openAIKey: process.env.SECRET_OPENAI_API_KEY
+       ```
+
+       When you debug the app, the Open AI key defined here process the Open AI key from the `.env.local.user` during local debug.
+
+    1. Go to the `teamsapp.local.yml`file and add the following code after `BOT_ID` and `BOT_PASSWORD`:
+
+       ```teamsapp.local.yml
+       SECRET_OPENAI_API_KEY: ${{SECRET_OPENAI_API_KEY}}
+       ```
+
+       When you run the app locally, the Open AI key defined here is used in the local settings.
+
+    1. Go to the `index.ts` file, and add the following code:
+
+        1. Add the following code to import Teams AI library library:
+
+            ```typescript
+            import { Application, ConversationHistory, DefaultPromptManager, DefaultTurnState, OpenAIModerator, OpenAIPlanner, AI } from '@microsoft/botbuilder-m365';
+            import path from "path";
+            ```
+
+            You need to import the required classes to use the functions and variables available in the class to build your app. You can leverage pre-built templates, and simply add your business logic to this scaffold.
+
+        1. Add the following code to create planner, moderator, and promptManager:
+
+            ```typescript
+            // Create AI components
+            const planner = new OpenAIPlanner({
+            apiKey: config.openAIKey,
```

Comment:
for proper formatting the JS object properties need to be indented with one "tab". Ex:
<img width="148" alt="image" src="https://github.com/MicrosoftDocs/msteams-docs/assets/115390646/33829d94-1d79-47cf-aca6-e811d4bfab0b">

This applies to the `moderator` and `promptManager` initialization as well.

Reply:
Updated.